eng/scripts/packagevalidation/PrePublish-Validation.ps1

```diff
@@ -0,0 +1,97 @@
+[CmdletBinding()]
+Param (
+  [ValidateNotNullOrEmpty()]
+  [string] $PackagePath,
+  [string] $ArtifactName,
+  [string] $WorkingDirectory
+)
+
+. "${PSScriptRoot}\..\common\scripts\logging.ps1"
+. "${PSScriptRoot}\..\common\scripts\SemVer.ps1"
+
+$WinVersion = (Get-ChildItem -Path "C:\Program Files (x86)\Windows Kits\10\bin\10.0*" | Sort-Object -Descending)[0]
+$SignTool = "C:\Program Files (x86)\Windows Kits\10\bin\$($WinVersion.Name)\x64\signtool.exe"
+
+New-Item -Path $WorkingDirectory -Name "Validation" -ItemType "directory"
+$ValidationDirectory = "$WorkingDirectory\Validation"
+
+Copy-Item -Path $PackagePath -Destination "$ValidationDirectory\package.zip"
+Expand-Archive -LiteralPath "$ValidationDirectory\package.zip" "$ValidationDirectory\extracted"
+
+$PackageName = Split-Path -Path $PackagePath -Leaf
+$PackageDlls = Get-ChildItem -Path "$ValidationDirectory\extracted\**\*.dll" -Recurse
+$NuspecFile = Get-ChildItem -Path "$ValidationDirectory\extracted\**\*.nuspec" -Recurse
+
+$NuspecContent = new-object xml
+$NuspecContent.Load($NuspecFile[0].FullName)
```

Comment:
We should do some validation to ensure there is exactly 1 nuspec.

Reply:
Should this also render the package invalid?

Reply:
Not sure but I don't just like to be careful that we aren't making bad assumptions.